### ISSUE TITLE
Automated (Detox) Testing - Implement Separate Feature Testing 

### ISSUE THREAD

**Issue by bischoffa** [department-of-veterans-affairs/va-mobile-app#8372]
# Project Status Overview

### Project Objective:

- Implement the ability to perform automated testing within individual features instead of having to run it for the entire app 
- Educate and train engineering how adding and updating detox test when code changes vs waiting for QA to do it

 
### Current Dependencies / Blockers
<details>
<summary>Blockers</summary>

|Blocker | Team / Owner | Est to Resolution |
|-------|---------|-------| 
| Internal dependency  | Internal PR review had been waiting for approval for week  | Following up 4/16 on how to close out this process | 
| Internal dependency | Engineers prioritizing this work | Completed - Tim said engineers will do this work once feature is complete, stable and they are trained | 
| External dependency ?  |   |
|  |   |

</details>

### Project Timeline
<details>
<summary>Milestones</summary>

|Step | Timing | 
|-------| ---------| 
| Implement separate feature detox testing |    |
| Create documentation on how this works |   |
| Determine what engineers will and won't do and when | | 
| Develop onboarding materials so that engineers learn how to do this  |   |
| Rollout responsibility change in having engineers handle this  |   |
| Determine how to oversee rollout and maintain engineers completing this task |  |
</details>

### Monthly Update for Stakeholders
<details>
<summary>Updates</summary>

|Month| Project Health | Progress & Key Accomplishments| Issues/Risks/Blockers|
|-------| ---------|--------|---------|
|Month Year | 
|Month Year| .....
|Month Year|
|Month Year|
</details>

# Detailed Project Scope 

### Problem Statement

- Currently  Automated Testing through the Detox Tool can only be run for the entire app, meaning if a change is made to a specific feature it has to be run for the whole app vs a specific piece of the app. This leads wasted time in running the detox because people are waiting on it testing items that are not warranted.  
- QA was told if we can get Detox to run for specific feature, then can have frontend engineering on the various mobile teams be responsible for creating or updating detox tests that align with the code changes they are making vs having only QA do it 
- Doing so will save time and effort and reduce the risks of errors 

### Assumptions 

<details>
<summary>Assumptions that need to be true for this project</summary>

 1.) Mobile Manager of Engineering still wants to have frontend do this work 
 
 2.) Frontend engineers can perform this work, can be trained, and routinely do this work 
 
 3.) 
 
 
</details>

### Risks 

<details>
<summary>Risks identified and accepted by stakeholders</summary>

 1.) FE will push back and not do the work or routinely forget 
 
 2.) 
 
 3.) 
 
 
</details>

### Definition of Done 
<details>
<summary>Conditions that must be true in order to consider this project completed</summary>

|Item | Completed | 
|-------| ---------|
| Implement separate feature detox testing |    |
| Create documentation on how this works |   |
| Determine what engineers will and won't do and when | | 
| Develop onboarding materials so that engineers learn how to do this  |   |
| Rollout responsibility change in having engineers handle this  |   |
| Determine how to oversee rollout and maintain engineers completing this task |  |

</details>

### Project Scope
<details>
<summary>Items in and out of scope</summary>

|In Scope | Out of Scope| 
|-------| ---------|
| | | 
| | | 

</details>

### Important Project Links 
<details>
<summary>Links</summary>
 
- Github Epic
- Github VA Mobile Page
- - Product Brief
- Product Canvas
- Design 
   - Discovery
   - Mockups
   - Final Design
   - Figma File
- Research
   - Research Plan
   - Research Results and Findings
-Analytics
-Decision Log 
</details>

### Team Members

<details>
<summary>Contact List</summary>

**VA:**
- **Mobile OCTO Product Leads:** Rachel Han or  Ryan Thurlwell
- **VA OCTO Product Leads:** 
- **External Team 1 & Role:**
- **External Team 1 & Role:**

**Ad Hoc:** 
- **Mobile Team**: Global, Health and Benefits, Design System, QA, API 
- **Mobile Team Lead:** 
- **FE Engineering:**
- **BE Engineering:** 
- **Design:**
- **Content**
- **QA:**
- **Mobile Teams that are involved:** Global, Health and Benefits, Design System, QA, API 

</details>

### Questions 
<details>
<summary>Open Questions</summary>
 
 - How are we measuring success? 
 
 - What is MVP and what is out of scope?
   
 - Has there been existing research or work done on this project before?
   
 - Is this work being done within the VA today (ex. web) 
 
</details>

<details>
<summary>Closed Questions</summary>

| Question |  Answer | Owner |
|--------|----------|------|
| | | |
| | | |

</details>

### Change / Decision Log 

Product will need to create on in the Github folder and link it here.

**Comment by bischoffa** [comment]
Separate epic but is part of the Qarts Bug and Maintenance epic / capacity planning

**Comment by TKDickson** [comment]
Closing this ticket - separate testing is possible, automated, and already done.
